Seems like 2.12.1 ignoring "ignore-paths" in config

### Bug description

Good day/morning/evening!

Today i've update pylint to version 2.12.1 and for now that version ignoring ignore-paths=.*/migrations/.*\.py,
in my .pylintrc config. But in version 2.11.1 everything is working correct

### Configuration

```ini
[MASTER]
ignore-paths=.*/migrations/.*\.py,.*/urls.py
disable=R0903,C0103,E0110,W0511,R0801,E1101
max-line-length=120
max-module-lines=800

[DESIGN]
max-attributes=30
max-parents=30
max-public-methods=50
```


### Command used

```shell
pylint --rcfile=.pylintrc apps
```


### Pylint output

```shell
************* Module apps.accounts.migrations.0001_initial
apps/accounts/migrations/0001_initial.py:27:0: C0301: Line too long (196/120) (line-too-long)
apps/accounts/migrations/0001_initial.py:28:0: C0301: Line too long (329/120) (line-too-long)
apps/accounts/migrations/0001_initial.py:29:0: C0301: Line too long (165/120) (line-too-long)
apps/accounts/migrations/0001_initial.py:30:0: C0301: Line too long (203/120) (line-too-long)
apps/accounts/migrations/0001_initial.py:38:0: C0301: Line too long (255/120) (line-too-long)
apps/accounts/migrations/0001_initial.py:45:0: C0301: Line too long (266/120) (line-too-long)
apps/accounts/migrations/0001_initial.py:46:0: C0301: Line too long (229/120) (line-too-long)
apps/accounts/migrations/0001_initial.py:1:0: C0114: Missing module docstring (missing-module-docstring)
apps/accounts/migrations/0001_initial.py:12:0: C0115: Missing class docstring (missing-class-docstring)
apps/accounts/migrations/0001_initial.py:9:0: C0411: standard import "import uuid" should be placed before "from django.conf import settings" (wrong-import-order)
************* Module apps.accounts.migrations.0002_alter_pacmanuser_last_login
apps/accounts/migrations/0002_alter_pacmanuser_last_login.py:1:0: C0114: Missing module docstring (missing-module-docstring)
apps/accounts/migrations/0002_alter_pacmanuser_last_login.py:6:0: C0115: Missing class docstring (missing-class-docstring)
************* Module apps.accounts.migrations.0003_alter_pacmanuser_username
apps/accounts/migrations/0003_alter_pacmanuser_username.py:1:0: C0114: Missing module docstring (missing-module-docstring)
apps/accounts/migrations/0003_alter_pacmanuser_username.py:6:0: C0115: Missing class docstring (missing-class-docstring)
************* Module apps.accounts.migrations.0004_auto_20211124_1539
apps/accounts/migrations/0004_auto_20211124_1539.py:20:0: C0301: Line too long (246/120) (line-too-long)
apps/accounts/migrations/0004_auto_20211124_1539.py:1:0: C0114: Missing module docstring (missing-module-docstring)
apps/accounts/migrations/0004_auto_20211124_1539.py:6:0: C0115: Missing class docstring (missing-class-docstring)
************* Module apps.accounts.migrations.0005_changeuseremailconfirmationcode
apps/accounts/migrations/0005_changeuseremailconfirmationcode.py:1:0: C0114: Missing module docstring (missing-module-docstring)
apps/accounts/migrations/0005_changeuseremailconfirmationcode.py:9:0: C0115: Missing class docstring (missing-class-docstring)
apps/accounts/migrations/0005_changeuseremailconfirmationcode.py:6:0: C0411: standard import "import uuid" should be placed before "from django.conf import settings" (wrong-import-order)
************* Module apps.notifications.migrations.0001_initial
apps/notifications/migrations/0001_initial.py:24:0: C0301: Line too long (385/120) (line-too-long)
apps/notifications/migrations/0001_initial.py:33:0: C0301: Line too long (130/120) (line-too-long)
apps/notifications/migrations/0001_initial.py:1:0: C0114: Missing module docstring (missing-module-docstring)
apps/notifications/migrations/0001_initial.py:10:0: C0115: Missing class docstring (missing-class-docstring)
apps/notifications/migrations/0001_initial.py:7:0: C0411: standard import "import uuid" should be placed before "from django.conf import settings" (wrong-import-order)
************* Module apps.notifications.migrations.0002_alter_notifications_notification_type
apps/notifications/migrations/0002_alter_notifications_notification_type.py:16:0: C0301: Line too long (518/120) (line-too-long)
apps/notifications/migrations/0002_alter_notifications_notification_type.py:1:0: C0114: Missing module docstring (missing-module-docstring)
apps/notifications/migrations/0002_alter_notifications_notification_type.py:6:0: C0115: Missing class docstring (missing-class-docstring)
************* Module apps.notifications.migrations.0003_alter_notifications_notification_type
apps/notifications/migrations/0003_alter_notifications_notification_type.py:16:0: C0301: Line too long (648/120) (line-too-long)
apps/notifications/migrations/0003_alter_notifications_notification_type.py:1:0: C0114: Missing module docstring (missing-module-docstring)
apps/notifications/migrations/0003_alter_notifications_notification_type.py:6:0: C0115: Missing class docstring (missing-class-docstring)
```


### Expected behavior

------------------------------------
Your code has been rated at 10.00/10


### Pylint version

```shell
pylint-2.12.1
```


### OS / Environment

Linux d507ea5a5499 5.14.17-301.fc35.x86_64 #1 SMP Mon Nov 8 13:57:43 UTC 2021 x86_64 GNU/Linux

### Additional dependencies

_No response_

--- a/ChangeLog
+++ b/ChangeLog
@@ -25,6 +25,10 @@ Release date: TBA
 * Fixed a false positive for ``unused-import`` where everything
   was not analyzed properly inside typing guards.
 
+* Specified that the ``ignore-paths`` option considers "\" to represent a
+  windows directory delimiter instead of a regular expression escape
+  character.
+
 ..
   Insert your changelog randomly, it will reduce merge conflicts
   (Ie. not necessarily at the end)

--- a/examples/pylintrc
+++ b/examples/pylintrc
@@ -22,7 +22,9 @@ ignore=CVS
 ignore-patterns=
 
 # Add files or directories matching the regex patterns to the ignore-list.
-# The regex matches against paths.
+# The regex matches against paths. NOTE: Because "\" represents the directory
+# delimiter on Windows systems it can't be used as the escape character
+# in these regular expressions.
 ignore-paths=
 
 # Python code to execute, usually for sys.path manipulation such as

